lib/elixir/lib/calendar/date.ex

```diff
@@ -398,6 +398,76 @@ defmodule Date do
     end
   end
 
+  @doc """
+  Converts a number of gregorian days to a `Date` struct.
+
+  ## Examples
+
+      iex> Date.from_gregorian_days(1)
+      {:ok, ~D[0000-01-02]}
+      iex> Date.from_gregorian_days(730_485)
+      {:ok, ~D[2000-01-01]}
+      iex> Date.from_gregorian_days(-1)
+      {:error, :invalid_days}
+
+  """
+  @spec from_gregorian_days(non_neg_integer(), Calendar.calendar()) :: {:ok, t} | {:error, atom}
+  def from_gregorian_days(days, calendar \\ Calendar.ISO)
+
+  def from_gregorian_days(days, calendar) when is_integer(days) and days >= 0 do
+    days
+    |> :calendar.gregorian_days_to_date()
```

Comment:
I thought one of the original design goals was to not rely on `:calendar` module?

Reply:
Wouldn't `from_iso_days/1` also support this without `:calendar` module required?

Reply:
Correct. We should implement this using the current from_iso_days function, doing a very quick conversion from iso days to gregorian days (I think we need to add or remove 365, i don't remember precisely).